composer/core/data_spec.py

```diff
@@ -12,43 +14,63 @@
 if TYPE_CHECKING:
     from composer.core.types import Batch, DataLoader
 
+__all__ = ["DataSpec"]
+
 
 class DataSpec:
-    """Specification for describing how to train and operate on data.
+    """Specifications for operating and training on data.
+
+    An example of constructing a :class:`DataSpec` object with a ``device_transforms`` callable
+    (:class:`~composer.datasets.utils.NormalizationFn`) and then using it with :class:`~.Trainer`:
+
+       >>> # In this case, we apply NormalizationFn 
+       >>> # Construct DataSpec as shown below to apply this transformation
+       >>> from composer.datasets.utils import NormalizationFn
+       >>> CHANNEL_MEAN = (0.485 * 255, 0.456 * 255, 0.406 * 255)
+       >>> CHANNEL_STD = (0.229 * 255, 0.224 * 255, 0.225 * 255)
+       >>> device_transform_fn = NormalizationFn(mean=CHANNEL_MEAN, std=CHANNEL_STD)
+       >>> dspec = DataSpec(train_dataloader, device_transforms=device_transform_fn)
```

Comment:
For this example, to follow what imagenet actually does, could the eval dataloader also be a dataspec and use the same normalization function?

Reply:
It can be used for eval_dataloader as well. Added a comment + DataSpec for eval in the example.